.github/workflows/build-wheels.yml

```diff
@@ -244,14 +247,24 @@ jobs:
 
       - name: merge wheels
         run: |
-          # unpack all single torch versions wheels in the same directory,
-          # and repack the directory as a new wheel
+          # collect all torch versions used for the build
+          REQUIRES_TORCH=$(find dist -name "*.whl" -exec unzip -p {} "metatensor_torch-*.dist-info/METADATA" \; | grep "Requires-Dist: torch")
+          MERGED_TORCH_REQUIRE=$(python scripts/create-torch-versions-range.py "$REQUIRES_TORCH")
+
+          echo MERGED_TORCH_REQUIRE=$MERGED_TORCH_REQUIRE
+
+          # unpack all single torch versions wheels in the same directory
+          mkdir dist/unpacked
+          find dist -name "*.whl" -print -exec python -m wheel unpack --dest dist/unpacked/ {} ';'
 
-          mkdir unpacked
-          find . -name "*.whl" -print -exec python -m wheel unpack --dest unpacked/ {} ';'
+          sed -i "s/Requires-Dist: torch ==.*/$MERGED_TORCH_REQUIRE/" dist/unpacked/metatensor_torch-*/metatensor_torch-*.dist-info/METADATA
 
+          echo "\n\n METADATA = \n\n"
+          cat dist/unpacked/metatensor_torch-*/metatensor_torch-*.dist-info/METADATA
```

Comment:
Do we want to keep thos debug information?

I think it is useful but just to be sure.

Reply:
yes, this is just printed in the CI so it might be worth keeping around.